docs/apis/plugintypes/qbank/filters.md

```diff
@@ -0,0 +1,215 @@
+        return [
+            datafilter::JOINTYPE_ANY,
+            datafilter::JOINTYPE_NONE,
+        ];
+    }
+```
+
+#### Custom filter class
+
+By default, the filter will be displayed and processed using the `core/datafilter/filtertype` JavaScript class.
+This will provide a single autocomplete field for selecting one or multiple numeric IDs with textual labels.
+If this does not fit your filter's use case, you will need to define your own filter class.
+
+Create a new JavaScript file in your plugin under `amd/src/datafilter/filtertypes/myfilter.js`.
+In this file, export a default class that extends `core/datafilter/filtertype`
+(or another core filter type from '/lib/amd/src/datafilter/filtertypes') and override the base methods as required.
+For example, if your filter uses textual rather than numeric values, you can override `get values()` to return the raw values
+without running `parseInt()` (see `types` filter). If you want a different UI for selecting your filter values instead of a
+single autocomplete, you can override `addValueSelector()`.
+
+To tell your filter condition to use a custom filter class, override the `get_filter_class()` method to return the namespaced
+path to your JavaScript class.
+
+```php
+    public function get_filter_class(): string {
+        return 'qbank_myplugin/datafilter/filtertype/myfilter';
+    }
+```
+
+#### Allow multiple values?
+
+By default, conditions allow multiple values to be selected and use the selected join type to decide how they are applied.
+If your condition should only allow a single value at a time, override `allow_multiple()` to return false.
+
+```php
+    public function allow_multiple(): bool {
+        return false;
+    }
+```
+
+#### Allow empty values?
+
+By default, conditions can be left empty, and therefore will not be included in the filter. To make it compulsory to select a
+value for this condition when it is added, override `allow_empty` to return false.
+
+```php
+    public function allow_empty(): bool {
+        return false;
+    }
+```
+
+#### Is the condition required?
+
+If it is compulsory that your condition is always displayed, override `is_required` to return true.
+
+```php
+    public function is_required(): bool {
+        return true;
+    }
+```
+
+#### Filter options
+
+If your condition supports additional options as to how the selected values are applied to the query, such as whether child
+categories are included when parent categories are selected, you can define "Filter options".
+
+In your condition class, define `get_filteroptions()` which returns an object containing the current filter options. You will
+probably want to add some code to the constructor to read in the current filter options, and some code the `build_query_from_filter`
+to use the option. See
+[`qbank_managecategories\category_condition`](https://github.com/moodle/moodle/blob/main/question/bank/managecategories/classes/category_condition.php#L331)
```

Comment:
Consider making this link point to a specific commit hash, or to just the file as a whole?